languages/csharp/exercises/concept/resource-cleanup/ResourceCleanupTests.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Reflection;
+using Xunit;
+using template;
+
+public class ResourceCleanupTests
+{
+    [Fact]
+    public void Write_good()
+    {
+        var db = new Database();
+        var orm = new Orm(db);
+        orm.Write("good write");
+        object[] actual = {db.DbState, db.lastData};
+        Assert.Equal(new object[] { Database.State.DataWritten, "good write"}, actual);
+    }
+
+    [Fact /*(Skip = "Remove this Skip property to run this test")*/]
+    public void Write_bad()
+    {
+        var db = new Database();
+        var orm = new Orm(db);
+        orm.Write("bad write");
+        object[] actual = {db.DbState, db.lastData};
+        Assert.Equal(new object[] { Database.State.Closed, "bad write"}, actual);
+    }
+
+    [Fact /*(Skip = "Remove this Skip property to run this test")*/]
+    public void Commit_good()
+    {
+        var db = new Database();
+        var orm = new Orm(db);
+        orm.Write("good commit");
+        orm.Commit();
+        object[] actual = {db.DbState, db.lastData};
+        Assert.Equal(new object[] { Database.State.Closed, "good commit"}, actual);
+    }
+
+    [Fact /*(Skip = "Remove this Skip property to run this test")*/]
+    public void Commit_bad()
+    {
+        var db = new Database();
+        var orm = new Orm(db);
+        orm.Write("bad commit");
+        orm.Commit();
+        object[] actual = {db.DbState, db.lastData};
+        Assert.Equal(new object[] { Database.State.Closed, "bad commit"}, actual);
+    }
+
+    [Fact /*(Skip = "Remove this Skip property to run this test")*/]
+    public void Disposable()
+    {
+        var db = new Database();
+        var orm = new Orm(db);
+        orm.Write("good data");
+        bool disposable = orm is IDisposable;
+        if (disposable)
+        {
+            typeof(Orm).InvokeMember("Dispose",
+                BindingFlags.Public | BindingFlags.Instance | BindingFlags.InvokeMethod,
+                null, orm, null);
+        }
```

Comment:
I think you could also do something like the following:

```suggestion
        var disposable = Assert.IsAssignableFrom<IDisposable>(orm);
        disposable.Dispose();
```

This will fail if `orm` is not implementing `IDisposable`, and if it is, it returns the object as an `IDisposable`.

Reply:
All is clear.  I missed a trick

I assume you are giving yourself a pass on he double assert :-)